File: Frontend/src/types/Receive/QuestDTO.ts
import type { QuestStatusType } from "./QuestStatusType";

export type QuestDTO = {
  id: string; 
  title: string; 
  description: string; 
  dateCreated: Date; 
  scheduledDate: Date; 
  dateCompleted?: Date | null; 
  isCompleted: boolean; 
  rewardPoints: number;
  userId: string; 
  hasActiveVote: boolean;
  status: QuestStatusType
};

File: API/QuestGiver/Models/Send/QuestDTO.cs
using QuestGiver.Data.Common;
using QuestGiver.Data.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace QuestGiver.Models.Send
{
    public class QuestDTO
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        [StringLength(30)]
        public string Title { get; set; }

        [Required]
        [StringLength(200)]
        public string Description { get; set; }

        public DateTime DateCreated { get; set; } = DateTime.UtcNow;
        public DateTime? DateCompleted { get; set; }
        public DateTime ScheduledDate { get; set; }

        [NotMapped]
        public bool IsCompleted => DateCompleted.HasValue; // Runtime calculated completion flag

        public QuestStatusType Status { get; set; }

        public bool HasActiveVote { get; set; } = false;

        [Required]
        public int RewardPoints { get; set; }

        [Required]
        [ForeignKey(nameof(User))]
        public Guid UserId { get; set; }
    }
}
